addons/knowledge/models/knowledge_article.py

```diff
@@ -438,10 +439,17 @@ def write(self, vals):
 
     @api.returns('self', lambda value: value.id)
     def copy(self, default=None):
+        self.ensure_one()
         self = self.sudo()
         default = dict(default or {},
                        name=_("%s (copy)", self.name),
-                       sequence=self.sequence+1)
+                       sequence=self.sequence + 1)
+        if self.user_has_access:
+            default.update({
+                "article_member_ids": [(0, 0, {
+                    'partner_id': self.env.user.partner_id.id,
+                    'permission': 'write'
+                })]})
```

Comment:
should check if the user is already a member. If it is, we should update it instead of adding it.